pkg/services/pluginsintegration/plugincontext/base_plugincontext.go

```diff
@@ -0,0 +1,62 @@
+package plugincontext
+
+import (
+	"context"
+	"runtime"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/useragent"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/auth/identity"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/adapters"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginconfig"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type BasePluginContextProvider interface {
+	// GetBasePluginContext returns a plugin context for the given plugin and user.
+	// It does not add DatasourceInstaceSettings or AppInstanceSettings
+	GetBasePluginContext(ctx context.Context, plugin pluginstore.Plugin, user identity.Requester) backend.PluginContext
+}
+
+func BaseProvideService(cfg *setting.Cfg, pluginRequestConfigProvider pluginconfig.PluginRequestConfigProvider) *BaseProvider {
```

Comment:
Nit. Suggest `ProvideBase` or `ProvideBaseService`. Since Providers and Provide Provider doesn't make since it make sense to use `NewBaseProvider` here instead.